fuel/schemes.py

```diff
@@ -232,3 +232,17 @@ def get_request_iterator(self):
         indices = list(self.indices)
         self.rng.shuffle(indices)
         return iter_(indices)
+
+@add_metaclass(ABCMeta)
+class LimitedScheme(IterationScheme):
+    """Iterate over the sequence given an iterationScheme object
+        but in a limited number of excursions defined by times
+    """
+    def __init__(self, iteration_scheme, times, *args, **kwargs):
+        self.iteration_scheme = iteration_scheme
+        self.times = times
+        assert self.times >=1, "Error : you need a positiv number of mini batches"
+
+    def get_request_iterator(self):
+        it = self.iteration_scheme.get_request_iterator()
+        return iter_([it.next() for n in range(self.times)])
```

Comment:
@vdumoulin You actually would have to do that for the result to be
picklable (generators are not). But you're better off just using
`picklable_itertools.islice`.

On Mon, Jun 22, 2015 at 7:18 PM, vdumoulin <notifications@github.com> wrote:

> In fuel/schemes.py
> <https://github.com/mila-udem/fuel/pull/177#discussion_r32995101>:
>
> > @@ -232,3 +232,17 @@ def get_request_iterator(self):
> >          indices = list(self.indices)
> >          self.rng.shuffle(indices)
> >          return iter_(indices)
> > +
> > +@add_metaclass(ABCMeta)
> > +class LimitedScheme(IterationScheme):
> > +    """Iterate over the sequence given an iterationScheme object
> > +        but in a limited number of excursions defined by times
> > +    """
> > +    def __init__(self, iteration_scheme, times, *args, **kwargs):
> > +        self.iteration_scheme = iteration_scheme
> > +        self.times = times
> > +        assert self.times >=1, "Error : you need a positiv number of mini batches"
> > +
> > +    def get_request_iterator(self):
> > +        it = self.iteration_scheme.get_request_iterator()
> > +        return iter_([it.next() for n in range(self.times)])
>
> Is there a reason why you're coercing it.next() for n in range(self.times)
> into a list?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mila-udem/fuel/pull/177/files#r32995101>.
>